src/main/java/Duke.java

```diff
@@ -1,10 +1,364 @@
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
 public class Duke {
     public static void main(String[] args) {
+
+        printWelcomeMessage();
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        System.out.println("loading stored data >>> Here is the content we found: ");
+        String home = System.getProperty("user.home");
+        java.nio.file.Path path = java.nio.file.Paths.get(home, "ip/src/main/java/DukeData.txt");
+        File f = new java.io.File(path.toUri());
+        try {
+            printFileContents(f);
+            addFileContents(f, tasks);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        System.out.println("____________________________________________________________");
+
+        while (true) {
+            String line = in.nextLine();
+            System.out.println("____________________________________________________________");
+            if ("bye".equalsIgnoreCase(line)) {
+                printByeMessage(f, tasks);
+                break;
+            }
+            try {
+                executeDukeCommands(line, tasks);
+            } catch (DukeException e) {
+                System.out.println("ERROR! Sorry, this is not a recognized Duke command!");
+            }
+            System.out.println("____________________________________________________________");
+        }
+    }
+
+    private static void printWelcomeMessage() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("____________________________________________________________");
+        System.out.println(" Hello! I'm Duke, your personal task manager.");
+        System.out.println(" What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
+
+    /**
+     * Reads contents of DukeData file and adds them to the Tasks ArrayList.
+     *
+     * @param f     The DukeData file holding the data.
+     * @param tasks The ArrayList holding the task objects.
+     * @throws IOException if there is some error in file reading.
+     */
+
+    private static void addFileContents(File f, ArrayList<Task> tasks) throws IOException {
+        Scanner s = new Scanner(f);
+        int index = 0;
+        while (s.hasNext()) {
+            String line = s.nextLine();
+            if (line.charAt(3) == 'T') {
```

Comment:
magic numbers here.
better to identify the position of the character by parsing rather than hardcoded indexing